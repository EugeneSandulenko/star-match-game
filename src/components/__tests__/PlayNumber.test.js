import PlayNumber from '../PlayNumber';
import renderer from 'react-test-renderer';
import React from 'react';
/* eslint-disable no-console */
describe('Test play number', () => {
  beforeEach(() => {
    // eslint-disable-next-line no-console
    console.log('Before each');
  });

  afterEach(() => {
    // eslint-disable-next-line no-console
    console.log('After each');
  });

  //   it('when play component is build then return button', () => {
  //     expect(2 + 2).toEqual(4);
  //   });

  //   it('async test', (done) => {
  //     setTimeout(done, 100);
  //   });

  function getRandomInt(max) {
    return Math.floor(Math.random() * Math.floor(max));
  }

  it('when number is provided then button display the number', () => {
    const number = getRandomInt(9) + 1;
    const tree = renderer.create(
      <PlayNumber status="available" number={number} />,
    );

    const instance = tree.root;
    const button = instance.findByProps({ className: 'number' });
    const text = button.children[0];

    expect(text).toEqual(String(number));
  });

  it('when status is available then button is rendered correctly', () => {
    const tree = renderer
      .create(<PlayNumber status="available" number={1} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('when status is invalid then button is rendered correctly', () => {
    const tree = renderer
      .create(<PlayNumber status="not_existing_status" number={1} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
