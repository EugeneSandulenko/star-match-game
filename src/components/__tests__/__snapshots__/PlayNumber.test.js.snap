// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test play number when status is available then button is lightgray 1`] = `
<button
  className="number"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  1
</button>
`;

exports[`Test play number when status is available then button is rendered correctly 1`] = `
<button
  className="number"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  1
</button>
`;

exports[`Test play number when status is invalid then button is lightgray 1`] = `
<button
  className="number"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  1
</button>
`;

exports[`Test play number when status is invalid then button is rendered correctly 1`] = `
<button
  className="number"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  1
</button>
`;

exports[`Test play number when status is invalid then button is undefined 1`] = `
<button
  className="number"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  1
</button>
`;
